### PR TITLE
removed RAG-related code sections

### PR DESCRIPTION
**Pull request by azheng1991**
Hi Nathan,

Here is the LLM-generated content for the bill before I removed the RAG-related code sections:

image

Here is the content after removing the RAG functions:

image

In my opinion, by making this change, the summary does seem a bit more straightforward and easy to digest. Also, the tagging appears to be more accurate as, on the back-end, it relies on GPT4 instead of the hard-coded list of tags that I manually expanded in last week's PR.  Maybe this is the best approach because by removing this RAG-related code, we are implicitly relying on the power of the LLM itself, which will only get better as time progresses. I was told recently that ChatGPT5 is already in the works in fact, so that is exciting. In the code below, I have tried to leave comments that explain the RAG-sections that were remove, although the diff also does a pretty good job of showing the changes. 

By the way, in my research of RAG, I came across the idea that it would be more beneficial latency-wise if we used a more direct approach, but in testing the application, I actually didn't really notice any marked differences in performance. Maybe if there are more bills though, the differences might be more observable.

### CODE REVIEW

File: demoapp/app2.py [hunk 1]
```diff
@@ -157,50 +157,37 @@ def generate_categories(text):
...
+    # Define the prompt template with placeholders for the text and category
     template = """You are a trustworthy assistant for question-answering tasks.
-        Use the following pieces of retrieved context to answer the question.
-        Question: {question}
-        Context: {context}
-        Answer:
-        """
-
-    prompt = PromptTemplate.from_template(template)
-    llm = ChatOpenAI(openai_api_key=API_KEY, temperature=0, model='gpt-4-1106-preview', model_kwargs={'seed': 42})
```
Comment (nesanders) [inline demoapp/app2.py:181]: Note that the old code was already using the GPT-4 model. I like the fact that the old code tagged a specific model version (`gpt-4-1106-preview`); I think this is preferable for reproducibility. Different model versions can yield significantly different results.

File: demoapp/app2.py [hunk 2]
```diff
@@ -157,50 +157,37 @@ def generate_categories(text):
...
-            # Do not output the tags outside from the list. \n
-            # """
-    with get_openai_callback() as cb:
-        response = rag_chain.invoke(query)
-        st.write(cb.total_tokens, cb.prompt_tokens, cb.completion_tokens, cb.total_cost)
+        Can you please explain what the following MA bill means to a regular resident without specialized knowledge? 
+        Please provide a one paragraph summary in 4 sentences. Be direct and concise:
+        {text}
         
+        After generating the summary, output the Category: {category}.
```
Comment (nesanders) [inline demoapp/app2.py:170]: Oddly, this prompt seems to try to do summariziation and categorization simultaneously. That came from the original code, I realize. I think we want one function that only does summarization and one that only does tagging.

File: demoapp/app2.py [hunk 3]
```diff
@@ -157,50 +157,37 @@ def generate_categories(text):
...
-            Summarize the bill that reads as follows:\n{text}\n\n
-            
-            After generating summary, output Category: {category}.
-            Then, output top 3 tags in this specific category from the list of tags {tags_for_bill} that are relevant to this bill. \n"""
-            # Do not output the tags outside from the list. \n
-            # """
-    with get_openai_callback() as cb:
-        response = rag_chain.invoke(query)
-        st.write(cb.total_tokens, cb.prompt_tokens, cb.completion_tokens, cb.total_cost)
+        Can you please explain what the following MA bill means to a regular resident without specialized knowledge?
```
Comment (nesanders) [inline demoapp/app2.py:166]: This version of the prompt is missing the extracted section text! Munira's PR adds this in, so it is critical to use.

**Review by nesanders — COMMENTED** [review]
Hi Allan,

Thanks very much for following up on this; I really apprecaite your willingness to engage with this LLM work and your taking on the task of stripping out the RAG implementation. 

But I think we should actually set aside this PR and refocus our efforts at our meeting tonight.

My reason for saying that is that we had talked about basing your branch on `mmomin15:munira/extractsections` rather than `main`, and this is important because Munira has introduced the code to integrate extracted section text in the query, which is a critical next step. This version does not have the section text, and so we do not expect it to perform well.

I have some other small comments I will publish to offer some general feedback.

**Comment by azheng1991** [comment]
@all-contributors please add @azheng1991 for <contributions>

**Comment by nesanders** [comment]
We decided this PR is obviated by Munira's other PR #5

**Pull request closed without merging**